pandas/core/arrays/interval.py

```diff
@@ -543,33 +543,81 @@ def __eq__(self, other):
             # extract intervals if we have interval categories with matching closed
             if is_interval_dtype(other_dtype):
                 if self.closed != other.categories.closed:
-                    return np.zeros(len(self), dtype=bool)
+                    return invalid_comparison(self, other, op)
                 other = other.categories.take(other.codes)
 
         # interval-like -> need same closed and matching endpoints
         if is_interval_dtype(other_dtype):
             if self.closed != other.closed:
-                return np.zeros(len(self), dtype=bool)
-            return (self._left == other.left) & (self._right == other.right)
+                return invalid_comparison(self, other, op)
+            if isinstance(other, Interval):
+                other = type(self)._from_sequence([other])
```

Comment:
Can't we use `other.left` / `other.right` scalars here? (and then the broadcasting for array vs scalar will work fine, and we don't have to deal with len-1 arrays?)